replica/table.cc

```diff
@@ -2848,8 +2848,12 @@ class compaction_group::table_state : public compaction::table_state {
             }
             return stop_iteration::no;
         };
-        if (main_sstable_set().for_each_sstable_until(check) == stop_iteration::no) {
-            maintenance_sstable_set().for_each_sstable_until(check);
+        if (!tables.empty()) {
+            std::any_of(tables.begin(), tables.end(), check);
```

Comment:
nit: I think `std::for_each()` (actually `std::ranges::for_each()`) would be a better fit here.